app/src/lib/git/reorder.ts

```diff
@@ -86,7 +86,7 @@ export async function reorder(
       // appeared on the log to reduce potential conflicts.
       if (beforeCommit !== null && commit.sha === beforeCommit.sha) {
         foundBaseCommitInLog = true
-        toReplayBeforeBaseCommit.unshift(commit)
+        toReplayAfterReorder.push(commit)
```

Comment:
I made a mistake here with my logic in #12357 which was putting the "base" commit before the commits you're moving. I adapted the tests in consequence to reflect the actual expected behavior (moved commits should be placed _before_ the base commit).